pkg/controllers/gateway/gateway_controller.go

```diff
@@ -94,12 +96,34 @@ func (r *GatewayReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 		}
 		return ctrl.Result{}, nil
 	}
+	if !controllerutil.ContainsFinalizer(previous, GatewayFinalizer) {
+		controllerutil.AddFinalizer(previous, GatewayFinalizer)
+		err = r.Update(ctx, previous)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+	}
 
 	if previous.GetDeletionTimestamp() != nil && !previous.GetDeletionTimestamp().IsZero() {
-		// TODO: Do we need to remove dns records and/or endpoints?
-		//       Will ownerRefs be sufficient
-		log.Info("Gateway is deleting", "gateway", previous.Name, "namespace", previous.Namespace)
+		log.Info("Deleting gateway", "gateway", previous.Name, "namespace", previous.Namespace)
+		//delete dnsRecord
+		err := r.DNSDeletion(ctx, *previous)
+		if err != nil {
+			log.Error(err, "Error deleting DNS record")
+		}
```

Comment:
What happens if you create 2 Gateways with the same host and delete one of them (The second one in particular)? Is it it still adding endpoints from both Gateways and then deleting the DNSRecord when either one of the Gateways is deleted?

Reply:
Yes exactly, with docs to come specifying that we do not support multiple gateways having the same hostname. A follow up ticket is to add some check to stop people from being able to implement it not just a warning in a doc